### DISCUSSION TITLE
How to check that element has either of classes in Cypress?

### DISCUSSION THREAD

**Discussion by ghost** [cypress-io/cypress#23859] [Component Testing]
I want to verify an element in such a way that it has one of 3 classes available. 

For example - cy.get('id') should have class1 **or** should have class2 **or** have class3

Below condition is not working
```
cy.get('id').should('have.class', '.class1')
      .or('have.class', '.class2')
     .or('have.class', '.class3')
```

**Comment by jjhelguero** [comment]
You can extract the `className` on your yielded element from the `.get()` and then use regex to match if it is either of your classes.

```js
cy.get("id").then(($el) => {
  const className = $el[0].className;
  expect(
    className,
    'element has either "class1", "class2", or "class3" class'
  ).to.match(/class1|class2|class3/);
});
```
or with more cypress commands
```js
cy.get("id")
  .then(($el) => $el[0].className)
  .should("match", /class1|class2|class3/);
```


You can find a working example [here](https://github.com/jjhelguero/cypress-examples/blob/main/cypress/e2e/checkElementHasEitherClassName.md).

**Reply by ghost** [reply]
Above solution is not working for my code:

HTML
```
<div class="item-head-wrapper">
     <div class="one">...</div>
     <div class="two">...</div>
     <div class="three">...</div>
     <div class="four">...</div>
</div>
```

:nth-child(4) will have either class="four", class="five" or class "six" sometime. 

`<div class="four" OR class="five" OR class="six">...</div>`

**Comment by nathan5x** [comment]
Another possible option is to use the `.satisfy` assertion. 
```
cy.get('id').should('satisfy', ($el) => {
   return $el.classList.contains('class1') || $el.classList.contains('class2') || $el.classList.contains('class3') ;
}};
```
